v2/apiserver/internal/core/kubernetes/substrate.go

```diff
@@ -568,3 +655,622 @@ func (s *substrate) countRunningPods(
+					"storage": storageQuantity,
+				},
+			},
+		},
+	}
+
+	pvcClient :=
+		s.kubeClient.CoreV1().PersistentVolumeClaims(project.Kubernetes.Namespace)
+	if _, err := pvcClient.Create(
+		ctx,
+		&workspacePVC,
+		metav1.CreateOptions{},
+	); err != nil {
+		return errors.Wrapf(
+			err,
+			"error creating workspace PVC for event %q worker",
+			event.ID,
+		)
+	}
+
+	return nil
+}
+
+func (s *substrate) createWorkerPod(
+	ctx context.Context,
+	project core.Project,
+	event core.Event,
+) error {
+	imagePullSecrets := []corev1.LocalObjectReference{}
+	if event.Worker.Spec.Kubernetes != nil {
+		for _, imagePullSecret := range event.Worker.Spec.Kubernetes.ImagePullSecrets { // nolint: lll
+			imagePullSecrets = append(
+				imagePullSecrets,
+				corev1.LocalObjectReference{
+					Name: imagePullSecret,
+				},
+			)
+		}
+	}
+
+	if event.Worker.Spec.Container == nil {
+		event.Worker.Spec.Container = &core.ContainerSpec{}
+	}
+	image := event.Worker.Spec.Container.Image
+	if image == "" {
+		image = s.config.DefaultWorkerImage
+	}
+	imagePullPolicy := event.Worker.Spec.Container.ImagePullPolicy
+	if imagePullPolicy == "" {
+		imagePullPolicy = s.config.DefaultWorkerImagePullPolicy
+	}
+
+	volumes := []corev1.Volume{
+		{
+			Name: "event",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: fmt.Sprintf("event-%s", event.ID),
+				},
+			},
+		},
+	}
+	if event.Worker.Spec.UseWorkspace {
+		volumes = append(
+			volumes,
+			corev1.Volume{
+				Name: "workspace",
+				VolumeSource: corev1.VolumeSource{
+					PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+						ClaimName: fmt.Sprintf("workspace-%s", event.ID),
+					},
+				},
+			},
+		)
+	}
+
+	volumeMounts := []corev1.VolumeMount{
+		{
+			Name:      "event",
+			MountPath: "/var/event",
+			ReadOnly:  true,
+		},
+	}
+	if event.Worker.Spec.UseWorkspace {
+		volumeMounts = append(
+			volumeMounts,
+			corev1.VolumeMount{
+				Name:      "workspace",
+				MountPath: "/var/workspace",
+				ReadOnly:  true,
+			},
+		)
+	}
+
+	initContainers := []corev1.Container{}
+	if event.Worker.Spec.Git != nil && event.Worker.Spec.Git.CloneURL != "" {
+		volumes = append(
+			volumes,
+			corev1.Volume{
+				Name: "vcs",
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{},
+				},
+			},
+		)
+
+		vcsVolumeMount := corev1.VolumeMount{
+			Name:      "vcs",
+			MountPath: "/var/vcs",
+		}
+
+		volumeMounts = append(volumeMounts, vcsVolumeMount)
+
+		initContainers = append(
+			initContainers,
+			corev1.Container{
+				Name: "vcs",
+				// TODO: For now, we're using the Brigade 1.x git init image, but for
+				// the sake of consistency and to lower the bar for creating additional
+				// VCS integrations, we should develop a new/improved image that gets
+				// input from a chunk of JSON, just like the actual worker image does.
+				Image:           "brigadecore/git-sidecar:v1.4.0",
```

Comment:
Same comment here about a constant for the string.

Would there be any point in just tagging `1.4.0` as `2.0.0-alpha` or something?

Reply:
Nice catch. I should add a TODO comment in here, as this is temporary.

There's an open issue for this. See #1134

Reply:
Wait... that comment already is a TODO.